cmd/storage-rest-client.go

```diff
@@ -160,14 +179,24 @@ func (client *storageRESTClient) Endpoint() Endpoint {
 	return client.endpoint
 }
 
-func (client *storageRESTClient) Healing() bool {
-	// This call should never be called over the network
-	// this function should always return 'false'
-	//
-	// To know if a remote disk is being healed
-	// perform DiskInfo() call which would return
-	// back the correct data if disk is being healed.
-	return false
+func (client *storageRESTClient) Healing() *healingTracker {
+	client.diskHealCache.Once.Do(func() {
+		// Update at least every second.
+		client.diskHealCache.TTL = time.Second
+		client.diskHealCache.Update = func() (interface{}, error) {
+			var h healingTracker
+			b, err := client.ReadAll(GlobalContext, minioMetaBucket,
+				pathJoin(bucketMetaPrefix, healingTrackerFilename))
+			if err != nil {
+				// If error, likely not healing.
+				return (*healingTracker)(nil), nil
+			}
+			_, err = h.UnmarshalMsg(b)
+			return &h, err
+		}
+	})
+	val, _ := client.diskHealCache.Get()
+	return val.(*healingTracker)
 }
```

Comment:
You have to aware that there is a potential for serializing the calls for disk.Healing() when Update() timing hits together i.e right at the time.Second'th 

The problem is also here that ReadAll() doesn't have a proper deadline such that the calls might just hang forever and serialize if the ReadAll() is blocked. Since Healing() function is going to be critical this path cannot hang.. 

We saw this problem with HTTP2 where the transport was hung

Reply:
Sure, I'll add a timeout.